packages/admin/dashboard/src/routes/locations/location-service-zone-shipping-option-create/components/create-shipping-options-form/create-shipping-option-details-form.tsx

```diff
@@ -52,6 +58,10 @@ export const CreateShippingOptionDetailsForm = ({
       })),
   })
 
+  useEffect(() => {
+    form.setValue("fulfillment_option_id", undefined)
+  }, [selectedProviderId])
+
```

Comment:
todo: this is missing a dependency, and we should set the value to "". To avoid using a useEffect, we could also update the provider_id select, to set `fulfillment_option_id` to `""` in the onChange.